backend/database/schema.sql

```diff
@@ -0,0 +1,110 @@
+create type role_type as enum ('admin', 'member');
+create type colour_type as enum ('red', 'orange', 'yellow', 'blue', 'green', 'grey', 'brown', 'purple', 'pink');
+create type checkout_type as enum ('borrow', 'use');
+create type checkout_status as enum ('waiting', 'checked_out', 'returned');
+create type approval_status as enum ('pending_approval', 'approved', 'not_approved');
+
+CREATE TABLE person (
+    zid             varchar(7) PRIMARY KEY,
+    password        varchar(12) NOT NULL,
+    first_name      varchar(30) NOT NULL,
+    last_name       varchar(30),
+    email           varchar(50) NOT NULL,
+    phone           varchar(10),
+    picture         text,
+    role            role_type NOT NULL,
+    CONSTRAINT CK_zid CHECK (zid ~* '^[1-9][0-9]{6}$'),
+    CONSTRAINT CK_email CHECK (email ~* '^[A-Za-z0-9]+[\._]?[A-Za-z0-9]+[@]\w+[.]\w{2,3}$'),
+    CONSTRAINT CK_phone CHECK (phone ~* '[0-9]{10}')
+);
+
+CREATE TABLE location (
+    id              serial PRIMARY KEY,
+    name            varchar(20) NOT NULL,
+    description     text,
+    picture         text
+);
+
+CREATE TABLE item (
+    sku             serial PRIMARY KEY,
```

Comment:
A SKU may contain letters (especially if using SKU values from other manufacturers)

eg. NVIDIA FE RTX3080 SKU: AC38220

If using SKU from other sources:
Consider having a separate ID for the item that's only used for DB purposes and not for users to see.

If SKU are uniquely created by the DB, you can ignore this.

Reply:
We're gonna maintain our own sku's I think, product sku's/UUID's may be kept in the description field